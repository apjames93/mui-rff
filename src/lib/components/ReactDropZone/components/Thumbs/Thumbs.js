import React from 'react';
import PropTypes from 'prop-types';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import Typography from '@material-ui/core/Typography';

const showPreview = (file) => {
  switch (file.type) {
    case 'application/pdf':
      return (
        <embed
          data-testid="embed"
          src={file.preview}
          type={file.type}
          alt="Preview"
          height="100%"
          width="100%"
        />
      );
    case 'application/json':
      return (
        <Typography variant="subtitle1" data-testid="json">
          {file.name}
        </Typography>
      );
    default:
      return (
        <img
          data-testid="image"
          type={file.type}
          src={file.preview}
          alt={file.name}
        />
      );
  }
};

const Thumb = ({
  files,
}) => (
  <GridList cellHeight={100} cols={3}>
    {files.length && files.map(file => (
      <GridListTile key={file.preview} cols={3}>
        {file.name && showPreview(file)}
      </GridListTile>
    ))}
  </GridList>
);
export default Thumb;

Thumb.propTypes = {
  files: PropTypes.arrayOf(PropTypes.shape({
    preview: PropTypes.string,
    name: PropTypes.string,
    type: PropTypes.string,
  })),
};

Thumb.defaultProps = {
  files: [],
};
