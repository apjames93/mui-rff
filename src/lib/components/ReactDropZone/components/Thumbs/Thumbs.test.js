import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Thumbs from './Thumbs';

function setup(type = '') {
  const props = {
    files: [{
      preview: 'preview',
      name: 'name',
      type,
    }],
  };
  const comp = mount(<Thumbs {...props} />);
  return { comp, props };
}

describe('<Thumbs />', () => {
  it('renders Thumbs', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('renders Thumbs image', () => {
    const { comp } = setup();
    const image = comp.find('[data-testid="image"]').length;
    expect(image).toEqual(1);
  });

  it('renders Thumbs json', () => {
    const { comp } = setup('application/json');
    const json = comp.find('[data-testid="json"]').length;
    expect(json).toEqual(3); // mui is passing data-testid down
  });

  it('renders Thumbs embed', () => {
    const { comp } = setup('application/pdf');
    const embed = comp.find('[data-testid="embed"]').length;
    expect(embed).toEqual(1);
  });


  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<Thumbs {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
