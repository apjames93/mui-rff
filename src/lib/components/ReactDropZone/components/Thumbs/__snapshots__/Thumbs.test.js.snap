// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbs /> snapshot 1`] = `
<ul
  className="MuiGridList-root"
  style={
    Object {
      "margin": -2,
    }
  }
>
  <li
    className="MuiGridListTile-root"
    style={
      Object {
        "height": 104,
        "padding": 2,
        "width": "100%",
      }
    }
  >
    <div
      className="MuiGridListTile-tile"
    >
      <img
        alt="name"
        data-testid="image"
        src="preview"
        type=""
      />
    </div>
  </li>
</ul>
`;
